src/curl.c

```diff
@@ -313,6 +314,18 @@ double swupd_curl_query_content_size(char *url)
 		return -1;
 	}
 
+	/* if the file doesn't exist in the remote server, the server will respond
+	 * with its "404 Not Found" page, which has a size itself (162 bytes for
+	 * NGINX which is the default content server used by clear). So if the file is
+	 * not found we need to return a size of 0 instead, otherwise the download size
+	 * calculation will be wrong */
+	curl_ret = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response);
+	if (curl_ret != CURLE_OK) {
+		return -1;
+	} else if (response == 404) {
```

Comment:
What if the response is a `403` or any other error code?

Should we set `CURLOPT_FAILONERROR` instead so that it's simpler to implement error handling for server error responses?

Reply:
That is a good point @phmccarty we can either set `CURLOPT_FAILONERROR` as you suggest or just check for a return code of 200, either way it is a good idea

Reply:
hmm actually as a second thought, I think it would be better to do something like this:
```
curl_ret = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response);
if (curl_ret != CURLE_OK) {
	return -1;
} else if (response == 404) {
	return 0;
} else if (response != 200) {
	return -1;
} 
```

this function is intended to only provide the size of the file to be downloaded so the total download size is calculated. If the download actually succeeds or not is handled in a different function. If the file does not exist (which is something that should never happen with full files but happens all the time with delta packs) is the only conditions that matters at this point, that is not an error we should just return 0. If another error happens which prevents us from retrieving the file size then we should return a failure to discard the download size calculation all together.

If we use `CURLOPT_FAILONERROR` then 404 is going to be treated as an error, and we don't want that.

Reply:
We handle 404 differently so we need to have fail_on_error == false as castulo said.